src/reporting/ReportingWindow.sol

```diff
@@ -90,19 +92,25 @@ contract ReportingWindow is DelegationTarget, Typed, Initializable, IReportingWi
         IMarket _market = IMarket(msg.sender);
         require(markets.contains(_market));
         IMarket.ReportingState _state = _market.getReportingState();
-        if (_state > IMarket.ReportingState.ALL_REPORTING) {
-            allReporterMarkets.remove(_market);
-            limitedReporterMarkets.remove(_market);
-            return false;
+
+        // If we recieve an automated report we may not need any reporting. We will uppdate the counter if the market proceeds to limited reporting
+        if (_state == IMarket.ReportingState.AUTOMATED_DISPUTE) {
+            totalMarketsThatRequiredReporting -= 1;
         }
-        if (_state > IMarket.ReportingState.LIMITED_REPORTING) {
+
+        if (_state != IMarket.ReportingState.ALL_REPORTING) {
```

Comment:
Consider inverting this if statement.  I find `!=` to be harder to mentally parse than `==`.

Reply:
Same with the one below.